### PR TITLE
feat(fast-sim): TextReporter — human-readable simulation output (#416)

### LINKED ISSUE #416: feat(fast-sim): TextReporter — human-readable simulation output
## Parent

Part of #413 (`:fast-sim` epic)

## Description

Implement `TextReporter` that subscribes to simulation events and prints human-readable text to stdout.

## SimulationEvent Data Class (new)

Format-agnostic event model enabling future JSON/CSV formatters:

```kotlin
data class SimulationEvent(
    val time: Double,
    val source: String,
    val type: String,     // ReportType name: TRAIN_EVENTS, NODE_EVENTS, PATH_SETTING, etc.
    val message: String
)
```

## Event Field Mapping

TextReporter receives `ContextChangeEvent` where:
- `event.propertyName` = `ReportType.name` (e.g., "TRAIN_EVENTS") → maps to `SimulationEvent.type`
- `event.newValue` = formatted string `"<simTime> <object> <message>"` (from `DefaultSimulationContext.report()` at line ~1338) → parsed into `SimulationEvent.time`, `.source`, `.message`

## TextReporter Design

Implements `ContextPropertyChangeListener`. Registered via `context.addPropertyChangeListener(reporter)` before `context.run()`.

**Output format:**
like in JVM

Summary line at end: `--- Simulation complete: <n> trains, <endTime>s sim time, <wallTime>s wall ---`

## Verbosity Flags

- **Default:** TRAIN_EVENTS + NODE_EVENTS + PATH_SETTING
- `--verbose`: adds TRAIN_CONTINUOUS (can produce thousands of lines)
- `--quiet`: summary line only

## Logging Strategy

**Research required**: Verify how kotlin-logging 7.0.3 configures log level on Kotlin/Native (no SLF4J backend). Check if `KotlinLoggingConfiguration` or equivalent API exists. Fallback options:
- Custom `KLogger` wrapper that discards below WARN
- Redirect kotlin-logging output to stderr via platform-specific native I/O
- Suppress all logging in fast-sim (set min level at initialization)

**Core requirement:** zero kotlin-logging output on stdout. TextReporter → stdout; logging → stderr.

## Tasks

- [ ] Implement `SimulationEvent` data class
- [ ] Implement `TextReporter` implementing `ContextPropertyChangeListener`
- [ ] Research kotlin-logging native log level configuration
- [ ] Implement `--verbose` / `--quiet` CLI flags
- [ ] Add summary line at simulation end
- [ ] Unit tests for event parsing and formatting

## Acceptance Criteria

- [ ] `./fast-sim example shuntingLoop 60` produces readable event output on stdout
- [ ] Zero kotlin-logging messages on stdout
- [ ] Default output excludes TRAIN_CONTINUOUS
- [ ] `--verbose` includes TRAIN_CONTINUOUS
- [ ] `--quiet` shows only summary
- [ ] Output parseable by regex `t=[\d.]+\s+\[\w+\]\s+.+`
- [ ] Summary line includes train count and wall-clock time

## Risk Register

| Risk | Severity | Mitigation |
|------|----------|------------|
| kotlin-logging stdout pollution | High | Research native log config API; fallback to custom wrapper or suppression |
| TRAIN_CONTINUOUS output flooding | Medium | Default exclude; `--verbose` opt-in |
| Event format parsing from report() | Low | Format is stable (established 2007, unchanged since) |

## LONG_TERM_GOALS Links

- **Goal 17 (Export)**: Same SimulationEvent model → CSV/Excel formatter
- **Goal 6 (Metrics)**: SimulationEvent = metrics collection foundation
- **Goal 9/10 (AI Dispatcher)**: Machine-parseable events for future agent monitoring

### PR DESCRIPTION
**Pull request by bedaHovorka**
## Summary
- Promote `SimulationEvent` from desktop-ui to `:core` commonMain with new `source` field, KMP-compatible `formatTime()` using `padStart`/`roundToInt`
- Add `TextReporter` (`ContextPropertyChangeListener`) in `:core` commonMain with `Verbosity` enum (QUIET/DEFAULT/VERBOSE) and injectable output lambda
- Wire into fast-sim CLI with `--verbose`/`--quiet` flags, suppress kotlin-logging on native via `KotlinLoggingConfiguration.logLevel = Level.OFF`
- Migrate desktop-ui to use shared `SimulationEvent`, delete old copy and `extractSimulationEvent()`

## Arg Handling Note

`--version` and `--help` are checked **position-independently** (anywhere in args). This was intentional to fix `fast-sim --verbose --help` not working. As a side effect, `fast-sim sim file.xml 60 --version` also exits early — acceptable, since these are meta-flags not simulation parameters.

## Test Plan
- [x] 16 SimulationEvent unit tests (formatTime edge cases, parsing, null handling)
- [x] 12 TextReporter unit tests (verbosity filtering, output format, summary, train counting with spaces)
- [x] 5 TextReporterIntegrationTest end-to-end tests (default/quiet/verbose modes, regex parsing, train count)
- [x] All existing desktop-ui EventTimelinePanelTest tests pass with migrated SimulationEvent
- [x] `./gradlew clean test` — all unit tests pass
- [x] `./gradlew integrationTest` — 150 integration tests pass
- [x] `./gradlew detekt ktlintCheck :core:checkCoreCommonMainPurity` — all quality gates pass
- [x] Manual binary verification: default (no TRAIN_CONTINUOUS), verbose (45 TRAIN_CONTINUOUS lines), quiet (summary only)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: fast-sim/src/linuxX64Main/kotlin/cz/vutbr/fit/interlockSim/fastsim/Main.kt [hunk 1]
```diff
@@ -61,14 +68,43 @@ fun main(args: Array<String>) {
...
 	val factory = NativeContextFactory()
 	val exitCode = try {
-		when (args[0]) {
-			CMD_EXAMPLE -> runExample(args, factory)
-			CMD_SIM     -> runSim(args, factory)
+		when (positionalArgs[0]) {
+			CMD_EXAMPLE -> runExample(positionalArgs, factory, verbosity)
+			CMD_SIM     -> runSim(positionalArgs, factory, verbosity)
 			else        -> { printUsage(); 2 }
 		}
```
[the remark this answers is not in the document]
Comment (bedaHovorka) [reply fast-sim/src/linuxX64Main/kotlin/cz/vutbr/fit/interlockSim/fastsim/Main.kt:122]: Fixed.  now checks `CMD_HELP in args` / `CMD_HELP_SHORT in args` instead of `args[0]`, so flags before `--help`/`--version` no longer hide them. Also added an `if (positionalArgs.isEmpty()) { printUsage(); 2 }` guard before the `when` block so a flags-only invocation like `fast-sim --verbose` prints usage and exits with code 2 instead of crashing.

File: fast-sim/src/linuxX64Main/kotlin/cz/vutbr/fit/interlockSim/fastsim/Main.kt [hunk 2]
```diff
@@ -45,14 +51,15 @@ private fun eprintln(message: String?) {
  *
  * @since Issue #415 (fast-sim native CLI)
  */
-@Suppress("TooGenericExceptionCaught")
-fun main(args: Array<String>) {
+/**
+ * Handles `--version`, `--help`, and empty args before Koin is started.
+ * Returns `null` when the program should continue, otherwise calls [exitProcess].
```
[the remark this answers is not in the document]
Comment (bedaHovorka) [reply fast-sim/src/linuxX64Main/kotlin/cz/vutbr/fit/interlockSim/fastsim/Main.kt]: Fixed. KDoc now reads: *"For these early-exit arguments, prints the appropriate output and calls `exitProcess`. If none of these arguments are present, returns normally and the program continues."*

File: core/src/commonMain/kotlin/cz/vutbr/fit/interlockSim/sim/SimulationEvent.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	val message: String
+) {
+	fun formatTime(): String {
+		val totalSeconds = simulationTime.toInt()
+		val hours = totalSeconds / 3600
+		val minutes = (totalSeconds % 3600) / 60
+		val seconds = totalSeconds % 60
+		val millis = ((simulationTime - totalSeconds) * 1000).roundToInt()
+		return "${hours.pad(2)}:${minutes.pad(2)}:${seconds.pad(2)}.${millis.pad(3)}"
+	}
```
[the remark this answers is not in the document]
Comment (bedaHovorka) [reply core/src/commonMain/kotlin/cz/vutbr/fit/interlockSim/sim/SimulationEvent.kt:21]: Fixed with `.coerceIn(0, 999)`. Added a test `formatTimeMillisDoesNotOverflow` that passes `simulationTime=1.9995` (which without the clamp would round to 1000) and asserts the output is `"00:00:01.999"`.

File: core/src/commonMain/kotlin/cz/vutbr/fit/interlockSim/sim/TextReporter.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		val wallSeconds = (currentTimeMillisKMP() - startWallTime) / 1000.0
+		val wallFormatted = ((wallSeconds * 10).toLong() / 10.0).toString()
+		output(
+			"--- Simulation complete: ${trainNames.size} trains, " +
+				"${lastSimTime}s sim time, ${wallFormatted}s wall ---"
+		)
+	}
+
+	private fun formatEvent(event: SimulationEvent): String =
+		"t=${event.simulationTime}  [${event.eventType.name}]  ${event.source} ${event.message}"
```
[the remark this answers is not in the document]
Comment (bedaHovorka) [reply core/src/commonMain/kotlin/cz/vutbr/fit/interlockSim/sim/TextReporter.kt]: Fixed. `formatEvent()` now uses the same conditional pattern as `formatForDisplay()`: if `source` is empty it emits `"$prefix  ${event.message}"`, otherwise `"$prefix  ${event.source} ${event.message}"`.
